internal/downloader/download.go

```diff
@@ -135,7 +168,31 @@ func (d *Downloder) download(desc, urlToGet, destination string, mode os.FileMod
 			fmt.Fprintln(os.Stderr, " done.")
 		}),
 	)
+	hasher := sha256.New()
+
+	_, err = io.Copy(io.MultiWriter(f, bar, hasher), resp.Body)
+	if err != nil {
+		return fmt.Errorf(
+			"Error while downloading text of %s into file %s: %v",
+			urlToGet, temporaryDestinationFile.Name(), err)
+	}
 
-	_, err = io.Copy(io.MultiWriter(f, bar), resp.Body)
-	return err
+	shaURLToGet := urlToGet + ".sha256"
+	shaExpected, err := d.getContentsOfURL(shaURLToGet)
+	if err != nil {
+		return fmt.Errorf("Error while trying to get contents of %s: %v", shaURLToGet, err)
+	}
+	shaExpected = strings.TrimRight(shaExpected, "\n")
+	shaActual := hex.EncodeToString(hasher.Sum(nil))
+	if shaExpected != shaActual {
+		return fmt.Errorf("Error downloading URL %s: Expected SHA [%s], got SHA [%s]",
+			urlToGet, shaExpected, shaActual)
+	}
+
+	tempInput, err := ioutil.ReadFile(temporaryDestinationFile.Name())
+	if err != nil {
+		return fmt.Errorf("Error reading temporary file %s: %v",
+			temporaryDestinationFile.Name(), err)
+	}
+	return ioutil.WriteFile(destination, tempInput, mode)
```

Comment:
I would not read the whole file and write it again into another location, instead I would simply move it, this is faster.

You can use something like
```golang
os.Rename(temporaryDestinationFile.Name(), destination)
os.Chmod(destination, 0755)
```

don't forget about checking the errors

Reply:
Done in 26008c3 - it looks like `os.Rename` uses hard links, which fail across devices, so we still need the old code as a backup